src/unified_diff.rs

```diff
@@ -116,11 +119,11 @@ where
     type Out = W;
 
     fn process_change(&mut self, before: Range<u32>, after: Range<u32>) {
-        if before.start - self.pos > 6 {
+        if before.start - self.pos > self.ctx_size {
```

Comment:
I did not understand why it would have to be `2 * self.ctx_size`. For me, we want to move `self.pos` to `before.start - self.ctx_size` if it's not already inside that range. But maybe I'm missing something, that will be clearer once I write some tests (or @pascalkuthe chimes in).

Reply:
I'd start at fixing CI before pinging people.

Reply:
OK so indeed `2 * self.ctx_size` works well when we are between 2 hunks. But for the first hunk it can be incorrect. For example, say you have `ctx_size=5`. First hunk starts at line 9, `self.pos` is currently at 0. Since `9 - 0 > 2 * 5`, you wont move `self.pos` and will print 8 lines of context before the first hunk.